meraki_config_template - Check for HTTP status code

kbreit opened the pull request:
##### SUMMARY
- All requests now check returned status code
- Fail if status code isn’t what is expected

Fixes #41282

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
meraki_config_template

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (meraki/2.7-http-code 88764165ad) last updated 2018/06/29 19:47:54 (GMT -500)
  config file = None
  configured module search path = ['/Users/kbreit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kbreit/Documents/Programming/ansible/lib/ansible
  executable location = /Users/kbreit/Documents/Programming/ansible/bin/ansible
  python version = 3.5.4 (default, Feb 25 2018, 14:56:02) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

kbreit committed:
Check for HTTP status code

- All requests now check returned status code
- Fail if status code isn’t what is expected

kbreit committed:
Fix blank line error

dagwieers commented on lib/ansible/modules/network/meraki/meraki_config_template.py:
```diff
@@ -83,7 +83,10 @@
-    return response
+    if meraki.status == 200:
+        return response
+    else:
```
Again, the else here is unneeded and unwanted.

dagwieers replied:
Looking at this again, I would prefer even this:
```python
if meraki.status != 200:
    meraki.fail_json(msg='Unable to get configuration templates')

return response
```

So that the default flow returns the response, except if the status is not 200. I think that's more clear, exceptions go into if-blocks, default flow is to succeed.

dagwieers commented on lib/ansible/modules/network/meraki/meraki_config_template.py:
```diff
@@ -111,7 +114,10 @@ def delete_template(meraki, org_id, name, data):
-    return response
+    if meraki.status == 200:
+        return response
+    else:
```
Again, the else here is unneeded and unwanted.

dagwieers commented on lib/ansible/modules/network/meraki/meraki_config_template.py:
```diff
@@ -126,7 +132,11 @@ def bind(meraki, org_name, net_name, name, data):
+        r = meraki.request(path, method='POST', payload=json.dumps(payload))
+        if meraki.status == 200:
+            return r
+        else:
```
Again, the else here is unneeded and unwanted.

dagwieers commented on lib/ansible/modules/network/meraki/meraki_config_template.py:
```diff
@@ -136,7 +146,11 @@ def unbind(meraki, org_name, net_name, name, data):
+        r = meraki.request(path, method='POST')
+        if meraki.status == 200:
+            return r
+        else:
```
Again, the else here is unneeded and unwanted.

dagwieers reviewed (changes requested):
I would change the default flow of functions to succeed, and make failures exceptions in if-blocks.

kbreit committed:
Change HTTP check logic and improve integration tests
- Set HTTP status code check so default path is accept
- Added create and delete network for integration test
- Remove a few comments to clean up code

kbreit reviewed (commented):
All changes made and submitted.

dagwieers reviewed (approved):
LGTM
